src/adapter.js

```diff
@@ -28,7 +28,9 @@ class MongooseAdapter {
     // by default, adapter is not filtered
     this.isFiltered = false;
 
-    mongoose.connect(uri, options);
+    mongoose.connect(uri, options).then(instance => {
```

Comment:
It is better to move this to `newAdapter` below. You can be sure that everything is ready, when connection established and appropriate event fired. Here is the example of code:

```js
newAdapter() {
  mongoose.connection.once('connected', () => {
    this.connection = connection;
  });
}
```